src/Middleware/Authentication/IGitHubAppTokenProvider.cs

```diff
@@ -0,0 +1,41 @@
+using System.Security.Cryptography;
+using Microsoft.IdentityModel.Tokens;
+
+namespace GitHub.Octokit.Client.Authentication;
+
+/// <summary>
+/// This is an interface for the provider that creates a token descriptor, uses that
+/// descriptor to create a JWT, and uses that JWT to create an access token
+/// for authentication as a GitHub App installation.
+/// </summary>
+public interface IGitHubAppTokenProvider
```

Comment:
Note that testing the logic of the `AppInstallationTokenProvider` was virtually impossible without a refactor. I chose to make the `GitHubAppTokenProvider` non-static, and provided this interface for it so we can easily mock its methods. That enables us to test `GetAuthorizationTokenAsync`'s functionality easily.

Reply:
That all makes sense. +1 on abstracting an interface as well for mocking... that was one thing I obviously left out in the implementation.